index-handlers/src/main/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandler.java

```diff
@@ -129,9 +143,12 @@ private void assertUserIsAllowedToSearchAffiliations(List<URI> affiliations, URI
         }
     }
 
-    private String extractQueryParamPublicationDateOrDefault(RequestInfo requestInfo) {
-        return requestInfo.getQueryParameters()
-            .getOrDefault(QUERY_PARAM_YEAR, String.valueOf(ZonedDateTime.now().getYear()));
+    private String extractQueryParamPublicationDateOrDefault(RequestInfo requestInfo, boolean isAdmin) {
+        var yearQueryParam =  requestInfo.getQueryParameters()
+            .getOrDefault(QUERY_PARAM_YEAR, null);
+        return isAdmin
+                   ? yearQueryParam
+                   : nonNull(yearQueryParam) ? yearQueryParam : String.valueOf(ZonedDateTime.now().getYear());
```

Comment:
admin can list all candidates not depending on year

Reply:
nested elvis (?:) are hard to read.
Will this do the same?

(isAdmin || nonNull(yearQueryParam))
  ? yearQueryParam
  : String.valueOf(ZonedDateTime.now().getYear())

Reply:
Also I feel like you are giving this method too much responibility compared with what the other "ExtractQueryParam**OrDefault" methods does. Could be better to just return the first value, and let the parent caller do the admin-check and decide which year to pass on